metrix/src/app.rs

```diff
@@ -41,6 +41,89 @@ fn query_metric_route(
 ) -> Result<Json<Vec<Metric>>, BadRequest<String>> {
     let db_conn = establish_connection();
 
+    let filter_clause: String;
+    let result = build_filter_clause(offset, start_datetime, end_datetime, q);
+    match result {
+        Ok(o) => {
+            filter_clause = o;
+        },
+        Err(e) => {
+            return Err(e);
+        },
+    }
+
+    println!("### QUERY: SELECT * FROM metrics {};", filter_clause);
+
+    let query_string = format!("SELECT * FROM metrics {} ORDER BY id LIMIT 10", filter_clause);
+    let results = sql_query(query_string)
+        .load(&db_conn)
+        .expect("Error loading metrics");
+
+    Ok(Json(results))
+}
+
+#[get("/<aggregation>?<offset>&<start_datetime>&<end_datetime>&<q>&<bucket_count>")]
+fn aggregate_metrics_route(
+    aggregation: Option<&RawStr>,
```

Comment:
~where do you use this?~ oh it's just `COUNT`.

I still think you should use it to turn it into a `COUNT`. But your call...

Reply:
I don't want to do that quite yet, since I haven't decided how I want to do SUM, AVG, etc which need to specify and actual field to operate on. Since count is a special case since it doesn't operate on a field, I figured it wouldn't make sense to address that initially